Policy-based governance, risk, and compliance paper (#94)
kubernetes/sig-security: sig-security-docs/papers/policy_grc/kubernetes-grc.md

* Reduces operational cost for SREs/Admins, enables continuous security readiness and audit readiness for enabled controls, and fosters day-to-day collaboration resulting in consensus among key personas, each with distinct business goals.

In larger organizations, the stakeholders for Kubernetes platforms may span several teams such as operations, development, security, compliance, finance, and indirectly customers as well as investors of the platform. Each of these teams will have different views of the system, and hence, different requirements for governance. A policy-based GRC approach allows stakeholders to collaborate effectively.

The diagram below shows how Kubernetes GRC builds on policy management and spans several functions. The following sections discuss each of these in detail.

![alt_text](images/kubernetes-policy-based-grc.png "Policy-Based Governance, Risk, and Compliance")

## Authors

* Anca Salier, IBM
* Andy Suderman, Fairwinds
* Gus Parvin, Red Hat
* Jayashree Ramanathan, Red Hat 
* Jim Bugwadia, Nirmata
* Poonam Lamba, Google
* Raj Krishnamurthy, ComplianceCow

> * Robert Ficcaglia, Sunstone Secure

Reviewer: ```suggestion
* Robert Ficcaglia, Sunstone Secure
* Poonam Lamba, ?
* Yuji Watanabe, ?
* Brandt Keller, ?
* Jon Zeolla, Seiso
```

Author: Thanks, updated.

* Expose or upload reports to GRC Tooling

# Tools

> **Note:** This section links to third party projects that provide functionality for Kubernetes. The Kubernetes project authors and CNCF aren't responsible for these projects, which are listed alphabetically. To add a project to this list, read the [content guide](https://kubernetes.io/docs/contribute/style/content-guide/#third-party-content) and [CNCF Website Guidelines](https://github.com/cncf/foundation/blob/main/website-guidelines.md) before submitting a change.

A community catalog of best practice policies can provide a starting point to implement policy-based GRC for the various domains discussed above. Several communities have formed to provide such policies for best practices in Kubernetes GRC, and enable the open sharing of contributions and improvements.

## Policy Engines
* [jsPolicy](https://www.jspolicy.com/): jsPolicy allows easier & faster Kubernetes Policies using JavaScript or TypeScript.
* [Kyverno](https://kyverno.io/): Kyverno is a policy engine designed for Kubernetes that allows validation, mutation, generation of resources and image verification, with no complex language necessary.

> * [KubeWarden](https://www.kubewarden.io/): Kubewarden is a policy engine for Kubernetes. It helps with keeping your Kubernetes clusters secure and compliant.

Reviewer 2: ```suggestion
* [Kubewarden](https://kubewarden.io/): Kubewarden is a policy engine for Kubernetes with validation, mutation, image verification, and context-aware policies. It aims to be the universal policy engine as its policies are Wasm modules, allowing to use the language of your choice.
```

* [OPA/Gatekeeper](https://open-policy-agent.github.io/gatekeeper/website/) provides admission control capability that can act on policies to check for conformance to best practices.
* [Polaris](https://www.fairwinds.com/polaris): Polaris is an open source policy engine for Kubernetes that validates and remediates resource configuration. It includes 30+ built in configuration policies, as well as the ability to build custom policies with JSON Schema.

## Runtime
* [Falco](https://falco.org/) : A cloud native runtime security tool that primarily monitors kernel events for unwanted activity on nodes
* [KubeArmor](https://kubearmor.io/): KubeArmor uses eBPF and Linux Security Modules (LSM) to provide policy based system to restrict any unwanted, malicious behavior of cloud-native workloads at runtime.

> * [Tetragon]() : Cilium’s new Tetragon component enables powerful real time, eBPF-based Security Observability and Runtime Enforcement.

Reviewer 2: ```suggestion
* [Tetragon](https://github.com/cilium/tetragon) : Cilium’s new Tetragon component enables powerful real time, eBPF-based Security Observability and Runtime Enforcement.
```

> * [Tracee]() : Tracee uses eBPF technology to tap into your system and give you access to hundreds of events that help you understand how your system behaves.

Reviewer 2: ```suggestion
* [Tracee](https://github.com/aquasecurity/tracee) : Tracee uses eBPF technology to tap into your system and give you access to hundreds of events that help you understand how your system behaves.
```

## Vulnerability Scanners
* [Grype](https://github.com/anchore/grype) : A container image, file system, and SBOM vulnerability scanner.
* [Trivy](https://github.com/aquasecurity/trivy) : A vulnerability scanner for container images, git repositories, file systems, kubernetes clusters, and more. Trivy can also be used to scan configuration and generate SBOMs.

## Configuration Scanners
* [Cloud Custodian](https://cloudcustodian.io/) : A rules engine that can detect, alert on, and/or modify resources that are misconfigured in public cloud accounts.
* [kube-bench](https://github.com/aquasecurity/kube-bench) : kube-bench is a tool that checks whether Kubernetes is deployed securely by running the checks documented in the CIS Kubernetes Benchmark.

> * [Kubescape](https://github.com/kubescape/kubescape) : Kubescape is an open-source Kubernetes security platform. It includes risk analysis, security compliance, and misconfiguration scanning

Reviewer 2: ```suggestion
* [Kubescape](https://github.com/kubescape/kubescape) : Kubescape is an open-source Kubernetes security platform. It includes risk analysis, security compliance, and misconfiguration scanning.
```

## SBOM Generators
* [OSS Review Toolkit](https://github.com/oss-review-toolkit/ort) - A suite of tools for managing open source projects including SBOM generation, dependency management, and technically enforcing the policies of a project.
* [sbom-tool](https://github.com/microsoft/sbom-tool) - Tool to create SPDX 2.2 compatible SBOMs for any variety of artifacts.
* [Syft](https://github.com/anchore/syft) - generates SBOMs from container images and filesystems.
* [Tern](https://github.com/tern-tools/tern) - A software composition analysis (SCA) tool that creates SBOMs for containers.

## Compliance Authoring
* [Trestle](https://ibm.github.io/compliance-trestle/) is an ensemble of tools that enable the creation, validation, and governance of documentation artifacts for compliance needs. It leverages NIST's OSCAL as a standard data format.

## Multi-cluster Governance

> * [Open Cluster Management](https://open-cluster-management.io/) : Provides a multi-cluster management technology that can be used  to manage multiple clusters for creating/importing clusters, configure them to desired state using policies

Reviewer 2: ```suggestion
* [Open Cluster Management](https://open-cluster-management.io/) : Provides a multi-cluster management technology that can be used  to manage multiple clusters for creating/importing clusters, configure them to desired state using policies.
```
